### ISSUE TITLE
Crash in 32 bits devices

### ISSUE THREAD

**Issue by felipe-valio-movile** [RevenueCat/purchases-ios#379]
I'am seeing a crash in iPad 4, iOS 10.3.4. This appears to be related to 32-bit based devices.

1. Environment
    1. Platform: iPad 4
    2. SDK version: 3.7.5
    3. OS version: 10.3.4
    4. Xcode version: 12.1
    5. How widespread is the issue: 100%
2. [Debug logs](https://docs.revenuecat.com/docs/debugging) that reproduce the issue
`Swift runtime failure: Not enough bits to represent the passed value [inlined]`

3. Steps to reproduce, with a description of expected vs. actual behavior
The bug is happening in production. I was able to reproduce it by compiling for release and making a sandbox subscription.

4. **Other information** 
<img width="1356" alt="Screen Shot 2020-10-28 at 09 07 45" src="https://user-images.githubusercontent.com/12138086/97439393-e192f400-1904-11eb-8b2d-f164b9bd4b4a.png">

```
Thread 1 Queue : com.apple.main-thread (serial)
#0  0x02425128 in Swift runtime failure: Not enough bits to represent the passed value [inlined] ()
#1  0x02425128 in specialized SignedInteger<>.init<A>(_:) [inlined] ()
#2  0x02425128 in ArraySlice<A>.toInt() [inlined] at /Users/felipe.valio/projetos/ios/vivo-meditacao-ios/Pods/PurchasesCoreSwift/PurchasesCoreSwift/LocalReceiptParsing/DataConverters/ArraySlice_UInt8+Extensions.swift:20
#3  0x02425128 in InAppPurchaseBuilder.build(fromContainer:) at /Users/felipe.valio/projetos/ios/vivo-meditacao-ios/Pods/PurchasesCoreSwift/PurchasesCoreSwift/LocalReceiptParsing/Builders/InAppPurchaseBuilder.swift:54
#4  0x02420418 in AppleReceiptBuilder.build(fromContainer:) at /Users/felipe.valio/projetos/ios/vivo-meditacao-ios/Pods/PurchasesCoreSwift/PurchasesCoreSwift/LocalReceiptParsing/Builders/AppleReceiptBuilder.swift:74
#5  0x0242e4ca in ReceiptParser.parse(from:) at /Users/felipe.valio/projetos/ios/vivo-meditacao-ios/Pods/PurchasesCoreSwift/PurchasesCoreSwift/LocalReceiptParsing/ReceiptParser.swift:47
#6  0x0242e218 in ReceiptParser.receiptHasTransactions(receiptData:) at /Users/felipe.valio/projetos/ios/vivo-meditacao-ios/Pods/PurchasesCoreSwift/PurchasesCoreSwift/LocalReceiptParsing/ReceiptParser.swift:32
#7  0x0242e2c4 in @objc ReceiptParser.receiptHasTransactions(receiptData:) ()
#8  0x023c44ce in __54-[RCPurchases restoreTransactionsWithCompletionBlock:]_block_invoke at /Users/felipe.valio/projetos/ios/vivo-meditacao-ios/Pods/Purchases/Purchases/Public/RCPurchases.m:594
#9  0x023c74dc in __30-[RCPurchases refreshReceipt:]_block_invoke at /Users/felipe.valio/projetos/ios/vivo-meditacao-ios/Pods/Purchases/Purchases/Public/RCPurchases.m:1005
#10 0x023ca296 in -[RCStoreKitRequestFetcher requestDidFinish:] at /Users/felipe.valio/projetos/ios/vivo-meditacao-ios/Pods/Purchases/Purchases/Purchasing/RCStoreKitRequestFetcher.m:134
#11 0x271a0192 in <redacted> ()
#12 0x03e56466 in _dispatch_call_block_and_release ()
#13 0x03e56452 in _dispatch_client_callout ()
#14 0x03e5ad46 in _dispatch_main_queue_callback_4CF ()
#15 0x1b94fd68 in <redacted> ()
#16 0x1b94de18 in <redacted> ()
#17 0x1b8a11ae in CFRunLoopRunSpecific ()
#18 0x1b8a0fd0 in CFRunLoopRunInMode ()
#19 0x1d04bb40 in GSEventRunModal ()
#20 0x20c23a52 in UIApplicationMain ()
#21 0x0000b352 in main at /Users/felipe.valio/projetos/ios/vivo-meditacao-ios/VivoMeditacao/App/AppDelegate.swift:17
#22 0x1b08e4ea in <redacted> ()
```

**Comment by aboedo** [comment]
great catch! the code seems to be assuming 64 bits. We'll get it fixed. Thanks for reporting!

**Comment by aboedo** [comment]
The fix has been merged and released in 3.7.6 🎉 
Thanks felipe-valio-movile for reporting!
